airflow/plugins_manager.py

```diff
@@ -30,6 +30,7 @@
 import importlib_metadata
 
 from airflow import settings
+from airflow.utils.entry_points_with_dist import entry_points_with_dist
```

Comment:
Module name is a little bit specific -- how about `from airflow.utils.plugins import entry_points_with_dist`?

Reply:
But I think "entry_point_with_dist" is a better name. we can use it also for non-plugins (providers_manager is technically not a plugin in the sense of "Airflow Plugin". This is a very nice a little small utility function that can be used for anything that wants to quickly retrieve entry_points, not necessarily plugins.

Reply:
My problem with calling the module `entry_points_with_dist`  is that it makes it a single use module -- we'd almost never be able to put anything else in here.

How about `airflow.utils.entry_points` then?

Reply:
That's fine, I can change it this way, no problem.

In general, my opinon is that there is nothing wrong in single class/function per module as long as there is nothing else it is really coupled with. 

I am not sure what benefits it could bring to group independent functions in the same module, other than accidental coupling. I prefer more files than unnecessary, accidental coupling. And you know - the design and coupling emerge over time and we could also rename it later (refactorings and renames are super easy if you use IDEs).

But I agree - having both module and function the same name is bad, so if only for that reason - happy to change it.